Purchases/Purchasing/StoreKitAbstractions/ProductType.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  ProductType.swift
+//
+//  Created by Nacho Soto on 2/15/22.
+
+import StoreKit
+
+extension StoreProduct {
+
+    /// The type of subscription of a product, whether a subscription or a one-time purchase.
+    ///
+    /// ### Related Symbols
+    /// - ``StoreProduct/ProductType-swift.enum``
+    @objc(RCStoreProductSubscriptionType)
+    public enum SubscriptionType: Int {
+
+        /// A non-renewable or auto-renewable subscription.
+        case subscription
+
+        /// A consumable or non-consumable in-app purchase.
+        case inAppPurchase
+
+    }
+
+    /// The type of product, equivalent to StoreKit's `Product.ProductType`.
+    ///
+    /// ### Related Symbols
+    /// - ``StoreProduct/SubscriptionType-swift.enum``
+    @objc(RCStoreProductType)
+    public enum ProductType: Int {
+
+        /// A consumable in-app purchase.
+        case consumable
+
+        /// A non-consumable in-app purchase.
+        case nonConsumable
+
+        /// A non-renewing subscription.
+        case nonRenewableSubscription
+
+        /// An auto-renewable subscription.
+        case autoRenewableSubscription
+
+    }
+
+}
+
+extension StoreProduct.ProductType {
+
+    var subscriptionType: StoreProduct.SubscriptionType {
+        switch self {
+        case .consumable: return .inAppPurchase
+        case .nonConsumable: return .inAppPurchase
+        case .nonRenewableSubscription: return .subscription
+        case .autoRenewableSubscription: return .subscription
+        }
+    }
+
+    /// Used as a placeholder when the type of product cannot be determined.
+    /// This value is considered undefined behavior.
+    static let defaultType: Self = .nonConsumable
```

Comment:
was this the default for any particular reason?

Reply:
Nope, just needed something. Like I tried to explain in the comment, I think this is better than making this optional everywhere.

Reply:
i guess the other option is a .unknown value. i feel like i'm behind on what we've been doing with other enums though, so i trust whatever you think is best

Reply:
This is only needed for SK1. If it was the other way around I'd say `.unknown` makes sense, but considering SK2 is the future, I think it's best to not have an extra case that would only be needed for SK1.